rpcs3/Emu/Cell/lv2/sys_rsx.cpp

```diff
@@ -414,12 +414,56 @@ error_code sys_rsx_context_iomap(cpu_thread& cpu, u32 context_id, u32 io, u32 ea
 		sys_rsx.warning("sys_rsx_context_iomap(): RSX is not idle while mapping io");
 	}
 
+	constexpr u32 _1m = 1u << 20;
+
+	std::unique_lock fast_lock(render->sys_rsx_mtx, std::defer_lock);
```

Comment:
I don't see the benefit of the double check under lock especially since we expect frame-to-frame the mappings wont actually change. Why not just lock and check once? I feel that would be faster here.